addons/sale_stock_margin/tests/test_sale_stock_margin.py

```diff
@@ -203,7 +203,11 @@ def test_so_and_multicurrency(self):
         other_currency = self.env.ref('base.EUR') if company_currency == self.env.ref('base.USD') else self.env.ref('base.USD')
 
         date = fields.Date.today()
-        ResCurrencyRate.create({'currency_id': company_currency.id, 'rate': 1, 'name': date})
+        today_rate = ResCurrencyRate.search([('name', '=', date), ('currency_id', '=', company_currency.id)])
+        if today_rate:
+            today_rate.rate = 1
+        else:
+            ResCurrencyRate.create({'currency_id': company_currency.id, 'rate': 1, 'name': date})
```

Comment:
Small review here!

I'm usually not a big fan of conditional test cases when it is possible to avoid them. There is another condition for the currency on the company.

Wouldn't it be possible to solve this issue by creating another company for the test? test_so_and_multicompany is actually creating one, this could maybe be done in a common setupclass :) And using this company for the test if possible.